modules/understanding-upgrade-channels.adoc

```diff
@@ -7,17 +7,129 @@
+available. Use this channel if you wish to receive updates as soon as they are
+available or for your pre-production environments when participating in the connected
+customer program. This channel will contain all z-stream ({product-version}.z)
+updates but will not suggest upgrades to the next minor release (4.5.z) when the next
+minor release is available. This channel also includes the upgrade from the previous
+4.3 release.
+
+= stable-{product-version}
+
+The stable channel, containing only GA releases, will contain updates on a time delay
+as they are gradually rolled out to customers based on data from our SRE teams,
+support services, and pre-production and production environments that participate in
+our connected customer program, rather than being immediately available as they are
+in the fast channel. For patch and CVE fixes this can range from several hours to a
+day and allows an extra period of assessment in how the software performs. This
+channel also includes the upgrade from the previous 4.3 release.
+
+= Fast and Stable Channel Usage and Strategies
+
+The fast-{product-version} and stable-{product-version} channels present a choice between receiving
+GA updates as soon as they are available or allowing Red Hat to control the rollout
+of those updates. If issues are detected during rollout, upgrades to that version may
+be blocked in both the fast and stable channels, and a new version may be introduced
+that will be the new preferred upgrade target.
+
+Customers can improve this process by configuring pre-production systems on the
+fast channel, production systems on the stable channel, and participating in Red
+Hat’s connected customer program. This program allows Red Hat to observe the impact
+of updates on your specific hardware and software configurations. Future releases may
+improve or alter the pace at which updates move from the fast to the stable channels.
+
+If issues are discovered with an upgrade between patch levels, Red Hat may withdraw
+that suggested upgrade for affected versions. A newer patch would become available
+in the appropriate channels and be suggested for upgrade.
+
+= Upgrade Version Paths
+
+{product-title} maintains an upgrade recommendation service that understands the
+version of {product-title} you have installed as well as the path to take within
+the channel you choose to get you to the next release. You can imagine seeing the
+following in the fast-{product-version} channel:
+
+* {product-version}.0
+* {product-version}.1
+* {product-version}.3
+* {product-version}.4
+
+The service only recommends upgrades that have been tested and have no known issues.
+If you are on {product-version}.1 and {product-title} is allowing you to select
+{product-version}.4, then it is safe for you to go from .{product-version}.1 to .{product-version}.4. Likewise,
+the absence of {product-version}.2 may be due to a CVE that was fixed in {product-version}.3 and Red Hat no
```

Comment:
Just not CVE, there might be some bugs which were discovered and applicable to some clusters

Reply:
Currently this example reads like "there was a CVE in {product-version}.2, so we dropped it from fast-{product-version}".  But we drop edges, not releases, when we discover new issues with existing releases.  We skip releases when we have build trouble (e.g. openshift/cincinnati-graph-data#33).  I'm fine talking about either or both of those, but I think it would help if this example graph also included edges.  Visual graphs are nice, like [the SVGs here][1], but I'd also be ok with:

```asciidoc
.  {product-version}.0, with recommended updates to {product-version}.3 and {product-version}.4
.  {product-version}.1, with recommended updates to {product-version}.4
.  {product-version}.3, with recommended updates to {product-version}.4
.  {product-version}.4
```

where an issue had been discovered with {product-version}.1 that lead us to pull its incoming edges.  Then we cut {product-version}.3 with improvements on {product-version}.0, but no safe {product-version}.1 -> {product-version}.3 update.  Then we cut {product-version}.4 with a safe update to get everyone off {product-version}.1.


[1]: https://github.com/openshift/cincinnati/blob/85c62120e5fc6779ed48540c9b06bf17e7652165/docs/design/openshift.md#channels